Tests/SwiftDocCTests/Infrastructure/DocumentationCuratorTests.swift

```diff
@@ -145,6 +145,96 @@ class DocumentationCuratorTests: XCTestCase {
         )
     }
     
+    func testModuleUnderTechnologyRoot() throws {
+        let (_, bundle, context) = try testBundleAndContext(copying: "ModuleUnderTechnologyRoot") { url in
+            try """
+            # Another Awesome Project
+            
+            Technology Root
+            
+            @Metadata {
+               @TechnologyRoot
+            }
+            
+            ## Overview
+            
+            This shouldn't create a warning because it's a technologyRoot
+            
+            ## Topics
+            
+            ### Code Reference
+            - ``MyAwesomeApp``
+            
+            """.write(to: url.appendingPathComponent("Root.md"), atomically: true, encoding: .utf8)
+        }
+        
+        var crawler = DocumentationCurator.init(in: context, bundle: bundle)
+        XCTAssert(context.problems.isEmpty, "Expected no problems. Found: \(context.problems.map(\.diagnostic.summary))")
+        
+        for node in context.rootModules {
+            XCTAssertNoThrow(try crawler.crawlChildren(of: node, prepareForCuration: { _ in }, relateNodes: { _, _ in }))
+        }
+        
+        XCTAssertEqual(crawler.problems.count, 0)
```

Comment:
This is only testing that there are no crawling warnings, which is already covered by checking `context.problems.isEmpty` since the content is already crawled while building up the documentation context.

It would be good to also check that the crawling formed the expected relationship between the technology root and the module by asserting the expected value of `context.pathsTo(/* the module reference */)`.

Reply:
I added a relationship check, but I'm not sure if it's the cleanest way to do so. Do you have any suggestions for how I could improve my code below?

```swift
guard let moduleNode = context.nodeWithSymbolIdentifier("SourceLocations"),
      let pathToRoot = context.pathsTo(moduleNode.reference).first,
      let root = pathToRoot.first else {
    
    XCTFail("Module doesn't have technology root as a predecessor in its path")
    return
}

XCTAssertEqual(root.path, "/documentation/Root")
```

I'm unsure if manually asserting `root.path` with a literal string is wise.

Reply:
This looks good to me. Since the root path is determined based on a value that comes from this test I don't see a problem with asserting on it as a string literal.